bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/HydrawiseCloudHandler.java

```diff
@@ -0,0 +1,246 @@
+            throws HydrawiseCommandException, HydrawiseConnectionException, HydrawiseAuthenticationException {
+        if (relay != null) {
+            client.stopRelay(relay.getRelayId());
+        }
+
+    }
+
+    @Override
+    protected void sendRunAllCommand()
+            throws HydrawiseCommandException, HydrawiseConnectionException, HydrawiseAuthenticationException {
+        client.runAllRelays(controllerId);
+    }
+
+    @Override
+    protected void sendRunAllCommand(int seconds)
+            throws HydrawiseCommandException, HydrawiseConnectionException, HydrawiseAuthenticationException {
+        client.runAllRelays(seconds, controllerId);
+    }
+
+    @Override
+    protected void sendStopAllCommand()
+            throws HydrawiseCommandException, HydrawiseConnectionException, HydrawiseAuthenticationException {
+        client.stopAllRelays(controllerId);
+    }
+
+    private void updateSensors(StatusScheduleResponse status) {
+        status.getSensors().forEach(sensor -> {
+            String group = "sensor" + sensor.getInput();
+            updateGroupState(group, CHANNEL_SENSOR_ACTIVE, sensor.getActive() > 0 ? OnOffType.ON : OnOffType.OFF);
+            updateGroupState(group, CHANNEL_SENSOR_MODE, new DecimalType(sensor.getType()));
+            updateGroupState(group, CHANNEL_SENSOR_NAME, new StringType(sensor.getName()));
+            updateGroupState(group, CHANNEL_SENSOR_OFFLEVEL, new DecimalType(sensor.getOfflevel()));
+            updateGroupState(group, CHANNEL_SENSOR_OFFTIMER, new DecimalType(sensor.getOfftimer()));
+            updateGroupState(group, CHANNEL_SENSOR_TIMER, new DecimalType(sensor.getTimer()));
+        });
+    }
+
+    private void updateForecast(StatusScheduleResponse status) {
+        int i = 1;
+        for (Forecast forecast : status.getForecast()) {
+            String group = "forecast" + (i++);
+            updateGroupState(group, CHANNEL_FORECAST_CONDITIONS, new StringType(forecast.getConditions()));
+            updateGroupState(group, CHANNEL_FORECAST_DAY, new StringType(forecast.getDay()));
+            updateGroupState(group, CHANNEL_FORECAST_HUMIDITY, new DecimalType(forecast.getHumidity()));
+            updateTemperature(forecast.getTempHi(), group, CHANNEL_FORECAST_TEMPERATURE_HIGH);
+            updateTemperature(forecast.getTempLo(), group, CHANNEL_FORECAST_TEMPERATURE_LOW);
+            updateWindspeed(forecast.getWind(), group, CHANNEL_FORECAST_WIND);
+        }
+    }
+
+    private void updateTemperature(String tempString, String group, String channel) {
+        Matcher matcher = TEMPERATURE_PATTERN.matcher(tempString);
+        if (matcher.matches()) {
+            try {
+                updateGroupState(group, channel, new QuantityType<Temperature>(Double.valueOf(matcher.group(1)),
+                        "C".equals(matcher.group(2)) ? SIUnits.CELSIUS : ImperialUnits.FAHRENHEIT));
+            } catch (RuntimeException e) {
```

Comment:
Please catch the `NumberFormatException `only, especially because you don't log the exception (stacktrace)

Reply:
Apply to the other places as well